src/js/fit/FIT.js

```diff
@@ -0,0 +1,389 @@
+    const fieldDefinitionNumber = blob[fieldDefinitionIndex];
+    /** Base type of the specified message's field (unsigned char, signed short, etc) */
+    const baseType = blob[fieldDefinitionIndex + 2];
+    /** Size (in bytes) of the specified message's field */
+    const size = blob[fieldDefinitionIndex + 1];
+
+    return {
+      baseType,
+      fieldDefinitionNumber,
+      size,
+      littleEndian: isLittleEndian,
+    };
+  }
+
+  readDefinitionMessage(blob, startIndex, localMessageType, hasDeveloperData) {
+    // Definition messages are used to associate local data message types to the global FIT message profile
+    // They describe the architecture, format, and fields of upcoming data messages
+
+    /** Architecture type */
+    const isLittleEndian = blob[startIndex + 2] === 0;
+    /** Number of fields in the data messages */
+    const numberOfFields = blob[startIndex + 5];
+    /** Number of Self Descriptive fields in the Data Message */
+    const numberOfDeveloperDataFields = hasDeveloperData ? blob[startIndex + 5 + numberOfFields * 3 + 1] : 0;
+    /** Global message number, unique to each message. Endianness of the 2 bytes value is defined in the architecture byte */
+    const globalMessageNumber = this.addEndian(isLittleEndian, [blob[startIndex + 3], blob[startIndex + 4]]);
+
+    const nextIndex = startIndex + 6 + (numberOfFields + numberOfDeveloperDataFields) * 3 + (hasDeveloperData ? 1 : 0);
+
+    this.messageTypes[localMessageType] = {
+      littleEndian: isLittleEndian,
+      globalMessageNumber,
+      fieldDefinitions: [
+        ...[...Array(numberOfFields).keys()].map((i) => this.readFieldDefinition(blob, startIndex, i, isLittleEndian)),
+        ...[...Array(numberOfDeveloperDataFields).keys()].map((i) =>
+          this.readDeveloperFieldDefinition(blob, startIndex, numberOfFields, i)
+        ),
+      ],
+    };
+
+    return {
+      messageType: MESSAGE_TYPE.DEFINITION,
+      nextIndex,
+    };
+  }
+
+  readDataMessage(blob, startIndex, recordHeader) {
+    const isCompressedTimestampDataMessage = this.isCompressedTimestampDataMessage(recordHeader);
+
+    /**
+     * The local message type is used to create an association between the definition message,
+     * data message and the FIT message in the Global FIT Profile.
+     *
+     * For a compressed timestamp header, only 2 bits are allocated for the local message type.
+     * As a result, the use of this special header is restricted to local message types 0–3,
+     * and cannot be used for local message types 4–15.
+     */
+    const localMessageType = isCompressedTimestampDataMessage
+      ? (recordHeader & COMPRESSED_LOCAL_MESSAGE_NUMBER_MASK) >> 5
+      : recordHeader & LOCAL_MESSAGE_TYPE_MASK;
+
+    /** Get the message type from the definitions */
+    const messageType = this.messageTypes[localMessageType];
+    if (!messageType) {
+      throw new Error('Missing message definition for local message number');
```

Comment:
sometimes you throw errors, sometimes you console.log and return empty array.
what will happen here on error, where is it caught?